server/game/core/ability/PlayCardAction.ts

```diff
@@ -1,16 +1,36 @@
 import * as CostLibrary from '../../costs/CostLibrary';
+import { resourceCard } from '../../gameSystems/GameSystemLibrary';
 import { IActionTargetResolver } from '../../TargetInterfaces';
 import { Card } from '../card/Card';
-import { PhaseName, PlayType, Stage } from '../Constants';
+import { KeywordName, PhaseName, PlayType, Stage } from '../Constants';
 import { ICost } from '../cost/ICost';
 import { AbilityContext } from './AbilityContext';
 import PlayerAction from './PlayerAction';
 
 export type PlayCardContext = AbilityContext & { onPlayCardSource: any };
 
 export abstract class PlayCardAction extends PlayerAction {
-    public constructor(card: Card, title: string, additionalCosts: ICost[] = [], targetResolver: IActionTargetResolver = null) {
-        super(card, title, additionalCosts.concat(CostLibrary.payPlayCardResourceCost()), targetResolver);
+    protected playType: PlayType;
+
+    public constructor(card: Card, title: string, playType: PlayType, additionalCosts: ICost[] = [], targetResolver: IActionTargetResolver = null) {
+        super(card, PlayCardAction.getTitle(title, playType), additionalCosts.concat(CostLibrary.payPlayCardResourceCost(playType)), targetResolver);
+
+        this.playType = playType;
+    }
+
+    /**
+     * Returns the appropriate title for the action based on the PlayType
+     * @param title The play title
+     * @param playType The play type
+     * @returns The play title for the action
```

Comment:
Not a change request, just a note that for really simple methods like this I'm fine omitting the detailed param / return descriptors. Totally fine if you prefer having them though, just a heads-up to potentially save time in the future